pytorch_lightning/utilities/debugging_examples.py

```diff
@@ -0,0 +1,300 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Debugging helpers meant to be used ONLY in the ``pl_examples`` and ``tests`` directories.
+
+Production usage is discouraged. No backwards-compatibility guarantees.
+"""
+import logging
+import os
+import random
+import time
+import urllib.request
+from typing import Optional, Tuple
+from urllib.error import HTTPError
+
+import torch
+from torch.utils.data import DataLoader, Dataset, Subset
+
+from pytorch_lightning import LightningDataModule, LightningModule
```

Comment:
I am not much happy about importing these high-level structures in utils, this may lead to import cycles...

Reply:
> this may lead to import cycles...

why? our source code should **never** import this. This is just for `tests` and `pl_examples`

Reply:
yes, ours not but how about users?

Reply:
I'm not sure what you mean. Users should never import these. See the top of the file

And even if they did, it wouldn't produce cyclic imports because these are only used in our tests and examples